pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -109,6 +118,54 @@ func fetchMetadata(endpoint string) (*caws.IIDAttestationData, error) {
 	}, nil
 }
 
+func getMetadataDoc(client *imds.Client) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), fetchMetadataDocTimeout)
+	defer cancel()
+	res, err := client.GetDynamicData(ctx, &imds.GetDynamicDataInput{
+		Path: docPath,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	doc, err := copyToString(res.Content)
+	if err != nil {
+		return "", err
+	}
+
+	return doc, nil
+}
+
+func getMetadataSig(client *imds.Client) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), fetchMetadataSigTimeout)
+	defer cancel()
+	res, err := client.GetDynamicData(ctx, &imds.GetDynamicDataInput{
+		Path: sigPath,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	sig, err := copyToString(res.Content)
+	if err != nil {
+		return "", err
+	}
+
+	return sig, nil
+}
+
+func copyToString(r io.ReadCloser) (res string, err error) {
```

Comment:
Can we name this something that makes it clear it takes "ownership" of the ReadCloser (i.e. closes it)?
```suggestion
func readStringAndClose(r io.ReadCloser) (res string, err error) {
```